app/Models/GenereRepository.php

```
@@ -424,6 +462,66 @@ public function merge(int $sourceId, int $targetId): array
+            // Preserve physical shelf placement: posizioni.genere_id is ON DELETE
+            // CASCADE and libri.posizione_id is ON DELETE SET NULL, so deleting the
+            // source genre would drop its posizioni and strip the shelf from every
+            // book placed there. Repoint the source genre's posizioni to the target
+            // instead. posizioni has a UNIQUE (scaffale_id, mensola_id, genere_id):
+            // when the target already owns a position on the same shelf+mensola,
+            // move the books onto that existing row and drop the duplicate source
+            // position rather than triggering a unique-key collision.
+            $srcPosStmt = $this->db->prepare("SELECT id, scaffale_id, mensola_id FROM posizioni WHERE genere_id = ?");
+            $srcPosStmt->bind_param('i', $sourceId);
+            if (!$srcPosStmt->execute()) {
+                throw new \RuntimeException('Errore nel recupero delle posizioni del genere di origine');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '476', column '27'). (undefined)

(MissingImport)